packages/webdriverio/src/utils/interception/devtools.js

```diff
@@ -107,7 +111,7 @@ export default class DevtoolsInterception extends Interception {
                         requestId,
                         responseCode,
                         responseHeaders,
-                        body: Buffer.from(newBody).toString('base64')
+                        body: newBody ? Buffer.from(newBody, 'binary').toString('base64') : undefined
```

Comment:
What happens if `newBody` is a string here?

Reply:
should be `undefined` https://github.com/puppeteer/puppeteer/blob/950ae334ca9c4c37b5a34dc61596a89596fd2bc6/src/common/HTTPRequest.ts#L381

Reply:
@christian-bromann got your point, I think.
There is no difference between an empty string and undefined in our case.
Let think fix that

Reply:
fixed